system/inc/system_sleep_configuration.h

```diff
@@ -71,16 +79,28 @@ class SystemSleepConfigurationHelper {
         auto wakeup = wakeupSourceFeatured(HAL_WAKEUP_SOURCE_TYPE_NETWORK);
         while (wakeup) {
             auto networkWakeup = reinterpret_cast<const hal_wakeup_source_network_t*>(wakeup);
-            if (networkWakeup->index == index) {
+            if (networkWakeup->index == index && !(networkWakeup->flags & HAL_SLEEP_NETWORK_FLAG_INACTIVE_STANDBY)) {
                 return true;
             }
             wakeup = wakeupSourceFeatured(HAL_WAKEUP_SOURCE_TYPE_NETWORK, wakeup->next);
         }
         return false;
     }
 
-    particle::EnumFlags<SystemSleepFlag> sleepFlags() const {
-        return particle::EnumFlags<SystemSleepFlag>::fromUnderlying(config_->flags);
+    EnumFlags<SystemSleepNetworkFlag> networkFlags(network_interface_index index) const {
+        auto wakeup = wakeupSourceFeatured(HAL_WAKEUP_SOURCE_TYPE_NETWORK);
+        while (wakeup) {
+            auto networkWakeup = reinterpret_cast<const hal_wakeup_source_network_t*>(wakeup);
+            if (networkWakeup->index == index) {
+                return EnumFlags<SystemSleepNetworkFlag>::fromUnderlying(networkWakeup->flags);
```

Comment:
Struct size needs to be checked. This new field may not be present for applications built with older DeviceOS versions.